GDJS/Runtime/pixi-renderers/pixi-json-manager.ts

```diff
@@ -65,36 +72,81 @@ namespace gdjs {
      * as JSON files can have been modified without the editor knowing).
      *
      * @param onProgress The function called after each json is loaded.
+     * @returns the promise of loaded jsons count.
      */
-    async preloadJsons(
-      onProgress: (loadedCount: integer, totalCount: integer) => void
+    async preloadAll(
+      onProgress: (loadingCount: integer, totalCount: integer) => void
     ): Promise<integer> {
-      const preloadedResources = [...this._resources.values()].filter(
-        (resource) => !resource.disablePreload
-      );
-
-      let loadedCount = 0;
-      await Promise.all(
-        preloadedResources.map(async (resource) => {
+      let loadedNumber = 0;
+      const getPreferences = (file: string) =>
+        ({
+          preferWorkers: false,
+          crossOrigin: this._resourcesLoader.checkIfCredentialsRequired(file)
+            ? 'use-credentials'
+            : 'anonymous',
+        } as Partial<PIXI.AssetsPreferences>);
+      const jsonPromises = Array.from(
+        this._resources.values(),
+        async (resource) => {
           try {
-            await this.loadJsonAsync(resource.name);
+            if (resource.kind === 'json') {
```

Comment:
Do you think it would be cleaner to create, in addition to the `atlas` resource, a `spine` resource type (even if it does contain JSON)?

Reply:
It is up to you. In this case would be great to have more complicated resource that has `spine` type and refers to three paths (json, atlas text and image files).

But creating of `spine` resource will help us to distinguish loading way in json manager or I even can restore old one and make `pixi-spine-manager` to load resources with `spine` type and build `SpineSkeleton` data by itself.